protocol/nats_jetstream/v3/message_test.go

```diff
@@ -190,3 +196,88 @@ func TestGetExtension(t *testing.T) {
 		})
 	}
 }
+
+func TestFinish(t *testing.T) {
+	type args struct {
+		err error
+	}
+	type wants struct {
+		err        error
+		ackCalled  bool
+		nackCalled bool
+	}
+	tests := []struct {
+		name  string
+		args  args
+		wants wants
+	}{
+		{
+			name: "nil error given",
+			args: args{
+				err: nil,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  true,
+				nackCalled: false,
+			},
+		},
+		{
+			name: "ACK error given",
+			args: args{
+				err: protocol.ResultACK,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  true,
+				nackCalled: false,
+			},
+		},
+		{
+			name: "NACK error given",
+			args: args{
+				err: protocol.ResultNACK,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  false,
+				nackCalled: true,
+			},
```

Comment:
nit: can you add cases for when the error is `jetstream.ErrMsgAlreadyAckd`

Reply:
added two more test conditions.